app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -1182,4 +1193,83 @@ public function getMagentoUrl($route = '', $params = array()){
         }
         return Mage::getUrl($route, $params);
     }
+
+
+    /**
+     *  Updates the shipping address data and, if necessary, the billing address data to the Magento
+     *  quote
+     *
+     * @param Mage_Sales_Model_Quote    $quote             The quote to which the address will be applied
+     * @param array                     $shippingAddress   The Bolt formatted address data
+     *
+     * @return  array   The shipping address applied in Magento compatible format
+     */
+    public function applyShippingAddressToQuote( $quote, $shippingAddress ) {
```

Comment:
`Api.php` is already too large. I suggest we create another helper class (eg., `QuoteHelper` to manipulate quote).

Reply:
Also suggest removing "Shipping" from method name as it would also change billing address

Reply:
Definitely.

Long-term, I believe that we are going to want to group this particular method under the "Shipping and Tax" umbrella.  Therefore, I will first propose this as a `ShippingAndTaxHelper`.

There are several methods in the ShippingController and the Api.php that I believe are destined to go here, but, that refactor will belong to a different task.

Reply:
@daisy1754 
> Also suggest removing "Shipping" from method name as it would also change billing address

I thought a good bit about this when naming the method and originally named it as you suggest.  However, what we are doing is always applying an address to the quotes shipping address.  The billing address is only applied shipping address data if it is missing this.  I have updated the methods doc to clarify this.  If you still think it is best to remove "Shipping" from the method name, then of course there is no problem with that.

_Note:  This is old logic that I don't necessarily agree with, (i.e. replacing parts of the billing address piecemeal.  I sense that it could eventually lead to an error of an incorrect billing address.  However, changing that logic would belong to a separate pull request_